content/en/docs/instrumentation/net/manual.md

```diff
@@ -0,0 +1,253 @@
+a console app or something that's ASP.NET Core-based.
+
+### Console app
+
+To start tracing in a console app, you need to create a tracer provider.
+
+First, ensure that you have the right packages:
+
+```
+dotnet add package OpenTelemetry
+dotnet add package OpenTelemetry.Exporter.Console
+```
+
+And then use code like this at the beginning of your program, during any important
+startup operations.
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Resources;
+
+// ...
+
+var serviceName = "MyServiceName";
+var serviceVersion "1.0.0";
+
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource(serviceName)
+    .SetResourceBuilder(
+        ResourceBuilder.CreateDefault()
+            .AddService(serviceName: serviceName, serviceVersion: serviceVersion))
+    .AddConsoleExporter()
+    .Build();
+
+//...
+```
+
+This is also where you can configure instrumentation libraries.
+
+Note that this sample uses the Console Exporter. If you are exporting to another endpoint,
+you'll have to use a different exporter.
+
+### ASP.NET Core
+
+To start tracing in an ASP.NET Core-based app, use the OpenTelemetry extensions for ASP.NET Core setup.
+
+First, ensure that you have the right packages:
+
+```
+dotnet add package OpenTelemetry --prerelease
+dotnet add package OpenTelemetry.Extensions.Hosting --prerelease
+dotnet add package OpenTelemetry.Exporter.Console --prerelease
+```
+
+And then configure it in your ASP.NET Core startup routine where you have access to an `IServiceCollection`.
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+// Define some important constants and the activity source
+var serviceName = "MyCompany.MyProduct.MyService";
+var serviceVersion = "1.0.0";
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Configure important OpenTelemetry settings, the console exporter, and automatic instrumentation
+builder.Services.AddOpenTelemetryTracing(b =>
+{
+    b
+    .AddConsoleExporter()
+    .AddSource(serviceName)
+    .SetResourceBuilder(
+        ResourceBuilder.CreateDefault()
+            .AddService(serviceName: serviceName, serviceVersion: serviceVersion))
+});
+```
+
+This is also where you can configure instrumentation libraries.
+
+Note that this sample uses the Console Exporter. If you are exporting to another endpoint,
+you'll have to use a different exporter.
+
+## Setting up an ActivitySource
+
+Once tracing is initialized, you can configure an `ActivitySource`, which will be how
+you trace operations with `Activity`s.
+
+Typically, an `ActivitySource` is instantiated once per app/service that is being instrumented,
+so it's a good idea to instantiate it once in a shared location. It is also typically named
+the same as the Service Name.
+
+```csharp
+using System.Diagnostics;
+
+public static class Telemetry
+{
+    //...
+
+    // 'TelemetryConstants.ServiceName' comes from somewhere in your code
```

Comment:
This applies to this and the earlier reference, but is this idiomatic .NET? Should the service name be set as a runtime constant or be loaded from config?

Reply:
I'll leave a common saying that it can also come from config. There's a ton of options available and it really just depends on what people prefer. Having a TelemetryConstants fie is likely to be an approach some folks take. But there's also others (e.g., appsettings.json)

Reply:
Sounds fine.